src/components/Article.jsx

```diff
@@ -1,83 +1,97 @@
 import { useParams } from "react-router-dom";
-import { getArticleById } from "../utils/api";
+import { getArticleById, incrementArticleVotes } from "../utils/api";
 import { useContext, useEffect, useState } from "react";
 import popSound from "../assets/popSound.mp3";
 
 import Comments from "./Comments";
 
-function popwithupvote() {
-  new Audio(popSound).play();
-
-  //to add vote functionality
-}
-function popwithdownvote() {
-  new Audio(popSound).play();
-
-  //to add vote functionality
-}
-
 function Article() {
   const { article_id } = useParams();
   const [loading, setLoading] = useState(true);
   const [error, setError] = useState(false);
   const [article, setArticle] = useState("");
   const [showComments, setShowComments] = useState(false);
+  const [displayedVotes, setDisplayedVotes] = useState(0);
+  const [newVote, setNewVote] = useState(1);
 
   function popwithcomments() {
     setShowComments(!showComments);
     new Audio(popSound).play();
   }
 
+  function popwithupvote() {
+    new Audio(popSound).play();
+    setDisplayedVotes(displayedVotes + 1);
+    setNewVote(1);
+    incrementArticleVotes(article_id, newVote).catch(() => {
+      setError(true);
+      setDisplayedVotes(displayedVotes - 1);
+    });
```

Comment:
A really great start with your optimistic rendering! You are setting the votes before you make your api request and reverting them back if there is an error 👍  
However, the voting is currently not quite working as the vote change is not persisting when the page is refreshed - this could be because of an issue with your patch request.